source/extensions/common/http_wasm/guest.h

```diff
@@ -0,0 +1,231 @@
+#pragma once
+
+#include "source/extensions/common/http_wasm/guest_config.h"
+#include "source/extensions/common/http_wasm/context.h"
+#include "source/extensions/common/http_wasm/vm_runtime.h"
+#include "source/extensions/common/http_wasm/bytecode_util.h"
+#include "envoy/server/lifecycle_notifier.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace HttpWasm {
+class GuestHandle;
+using EnvironmentVariableMap = std::unordered_map<std::string, std::string>;
+
+using WasmVmFactory = std::function<std::unique_ptr<Runtime>()>;
+using GuestHandleFactory = std::function<std::shared_ptr<GuestHandle>()>;
+using GuestHandleCloneFactory =
+    std::function<std::shared_ptr<GuestHandle>(std::shared_ptr<GuestHandle> wasm)>;
+using CallOnThreadFunction = std::function<void(std::function<void()>)>;
+
+class GuestConfig;
+class Guest : public Logger::Loggable<Logger::Id::wasm>,
+              public std::enable_shared_from_this<Guest> {
+public:
+  Guest(GuestConfig& config, const Stats::ScopeSharedPtr& scope, Api::Api& api,
+        Upstream::ClusterManager& cluster_manager, Event::Dispatcher& dispatcher);
+  Guest(std::shared_ptr<GuestHandle> other, Event::Dispatcher& dispatcher);
```

Comment:
q: is this constructor used?

Reply:
yes, for creating thread local guest from main guest